File: audit_images.py
import re
from pathlib import Path

root = Path(r'c:/Users/User/Giannis-main')
text = (root / 'categories.js').read_text(encoding='utf-8', errors='ignore')
refs = sorted(set(re.findall(r'"image"\s*:\s*"([^"]+)"', text)))
actual = []
for p in root.glob('images/**/*'):
    if p.is_file() and p.suffix.lower() in {'.png', '.jpg', '.jpeg', '.gif', '.webp', '.svg'}:
        actual.append(p.relative_to(root).as_posix())

actset = set(actual)
missing = [r for r in refs if not (root / r).exists()]
unref = [p for p in actset if p not in refs]


def norm(s: str) -> str:
    return re.sub(r'[^a-z0-9]+', '', s.lower())

similar = []
for r in missing:
    base = Path(r).name
    candidates = [a for a in actset if Path(a).name == base]
    if not candidates:
        candidates = [a for a in actset if norm(Path(a).name) == norm(base)]
        if candidates:
            similar.append((r, candidates))

print('TOTAL_REFERENCED_UNIQUE', len(refs))
print('TOTAL_ACTUAL_FILES', len(actset))
print('MISSING_REFERENCES', len(missing))
print('UNREFERENCED_FILES', len(unref))
print('POSSIBLE_RENAMES_OR_MISPLACED', len(similar))
print('MISSING_REFERENCES_LIST')
for item in missing:
    print(item)
print('UNREFERENCED_FILES_LIST')
for item in unref:
    print(item)
print('SIMILAR_NAME_MATCHES')
for ref, cands in similar:
    print(ref, '->', cands)

File: audit_images.js
const fs = require('fs');
const path = require('path');

const root = process.cwd();
const text = fs.readFileSync(path.join(root, 'categories.js'), 'utf8');
const refs = [...new Set(Array.from(text.matchAll(/"image"\s*:\s*"([^"]+)"/g), m => m[1]))].sort();

const actual = [];
function walk(dir) {
  for (const p of fs.readdirSync(dir, { withFileTypes: true })) {
    const full = path.join(dir, p.name);
    if (p.isDirectory()) walk(full);
    else if (/\.(png|jpg|jpeg|gif|webp|svg)$/i.test(p.name)) {
      actual.push(path.relative(root, full).split(path.sep).join('/'));
    }
  }
}
walk(path.join(root, 'images'));

const missing = refs.filter(r => !fs.existsSync(path.join(root, r)));
const unref = actual.filter(p => !refs.includes(p));
const norm = s => s.toLowerCase().replace(/[^a-z0-9]+/g, '');
const similar = [];
for (const r of missing) {
  const base = path.basename(r);
  const sameName = actual.find(a => path.basename(a) === base);
  if (!sameName) {
    const cands = actual.filter(a => norm(path.basename(a)) === norm(base));
    if (cands.length) similar.push([r, cands]);
  }
}

console.log('TOTAL_REFERENCED_UNIQUE', refs.length);
console.log('TOTAL_ACTUAL_FILES', actual.length);
console.log('MISSING_REFERENCES', missing.length);
console.log('UNREFERENCED_FILES', unref.length);
console.log('POSSIBLE_RENAMES_OR_MISPLACED', similar.length);
console.log('MISSING_REFERENCES_LIST');
for (const item of missing) console.log(item);
console.log('UNREFERENCED_FILES_LIST');
for (const item of unref) console.log(item);
console.log('SIMILAR_NAME_MATCHES');
for (const [ref, cands] of similar) console.log(ref, '->', cands.join(','));
